src/options.c

```diff
@@ -321,24 +321,54 @@ static struct opt_entry opt_list[] = {
 
 #if HAVE_LIBGMP || HAVE_INT128 || HAVE___INT128 || HAVE___INT128_T
 #define PRINCE_USAGE \
-"--prince[=FILE]            PRINCE mode, read words from FILE\n"
+"--prince[=FILE]            PRINCE mode, read words from FILE\n" \
+"--prince-loopback[=FILE]   fetch words from a .pot file\n" \
+"--prince-elem-cnt-min=N    minimum number of elements per chain (1)\n" \
+"--prince-elem-cnt-max=[N]  maximum number of elements per chain (negative N is\n" \
+"                           relative to word length) (8)\n" \
+"--prince-skip=N            initial skip\n" \
+"--prince-limit=N           limit number of candidates generated\n" \
+"--prince-wl-dist-len       calculate length distribution from wordlist\n" \
+"--prince-wl-max=N          load only N words from input wordlist\n" \
+"--prince-case-permute      permute case of first letter\n" \
+"--prince-mmap              memory-map infile (not available with case permute)\n" \
+"--prince-keyspace          just show total keyspace that would be produced\n" \
+"                           (disregarding skip and limit)\n"
 #else
 #define PRINCE_USAGE ""
 #endif
 
-#define JOHN_USAGE	  \
+#ifdef HAVE_FUZZ
+#define FUZZ_USAGE \
+"--fuzz[=DICTFILE]         fuzz formats' prepare(), valid() and split()\n" \
+"--fuzz-dump[=FROM,TO]     dump the fuzzed hashes between FROM and TO to file pwfile.format\n"
+#else
+#define FUZZ_USAGE ""
+#endif
+
+#define JOHN_BANNER	  \
 "John the Ripper " JTR_GIT_VERSION _MP_VERSION DEBUG_STRING ASAN_STRING UBSAN_STRING " [" JOHN_BLD "]\n" \
 "Copyright (c) 1996-2019 by " JOHN_COPYRIGHT "\n" \
 "Homepage: https://www.openwall.com/john/\n" \
 "\n" \
-"Usage: %s [OPTIONS] [PASSWORD-FILES]\n" \
+"Usage: %s [OPTIONS] [PASSWORD-FILES]\n"                                                  \
+
+#define JOHN_USAGE    \
+"--help                     print usage summary\n" \
+"--config=FILE              use FILE instead of john.conf or john.ini\n" \
```

Comment:
I'd move `--config` to at least after `JOHN_USAGE_REGEX` but probably much more down. Perhaps such shuffling (I can think of more) should be a separate issue and PR though.

Arguably we could also opt to list all options in alphabetical order.

Reply:
I put it just above logging and verbosity.